File: aula 3/repeticao.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int num = 0;
    vector<string> lista_de_produtos;

    while (num != 3) {
        cout << "\nDigite um dos números abaixo:\n1-Adicionar\n2-Listar\n3-Sair\n\n";
        cin >> num;

        if (num == 1) {
            string nome_do_produto;
            int quantidade_em_estoque;
            float preco_unitario;

            cout << "Digite o nome do produto: ";
            cin >> nome_do_produto;

            cout << "Digite a quantidade em estoque: ";
            cin >> quantidade_em_estoque;

            cout << "Digite o preço unitário: ";
            cin >> preco_unitario;

            if (quantidade_em_estoque < 0) {
                cout << "\nErro: quantidade negativa.\n";
            } else {
                string produto = nome_do_produto + " " + to_string(quantidade_em_estoque) + " " + to_string(preco_unitario);
                lista_de_produtos.push_back(produto);
                cout << "\nProduto cadastrado!\n";
            }
        }

        else if (num == 2) {
            for (int i = 0; i < lista_de_produtos.size(); i++) {
                cout << "\nProduto " << i+1 << ": " << lista_de_produtos[i] << endl;
            }
        }

        else if (num == 3) {
            break;
        }

        else {
            cout << "Opção inválida!\n";
        }
    }

    return 0;
}

File: aula 3/repeticao.java
import java.util.ArrayList;
import java.util.Scanner;

public class repeticao {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        int num = 0;
        ArrayList<String> lista_de_produtos = new ArrayList<>();

        while (num != 3) {
            System.out.println("\nDigite um dos números abaixo:\n1-Adicionar\n2-Listar\n3-Sair\n");
            num = sc.nextInt();

            if (num == 1) {
                System.out.print("Digite o nome do produto: ");
                String nome_do_produto = sc.next();

                System.out.print("Digite a quantidade em estoque: ");
                int quantidade_em_estoque = sc.nextInt();

                System.out.print("Digite o preço unitário: ");
                double preco_unitario = sc.nextDouble();

                if (quantidade_em_estoque < 0) {
                    System.out.println("\nErro: quantidade negativa.");
                } else {
                    String produto = nome_do_produto + " " + quantidade_em_estoque + " " + preco_unitario;
                    lista_de_produtos.add(produto);
                    System.out.println("\nProduto cadastrado!");
                }
            }

            else if (num == 2) {
                for (int i = 0; i < lista_de_produtos.size(); i++) {
                    System.out.println("\nProduto " + (i+1) + ": " + lista_de_produtos.get(i));
                }
            }

            else if (num == 3) {
                break;
            }

            else {
                System.out.println("Opção inválida!");
            }
        }
    }
}